satpy/readers/yaml_reader.py

```diff
@@ -718,6 +718,86 @@ def _load_dataset_area(self, dsid, file_handlers, coords):
                 logger.debug("No coordinates found for %s", str(dsid))
             return area
 
+    @staticmethod
+    def _area_def_coords(data_arr, area, crs=None):
+        """Assign x/y coordinates to DataArray from provided area.
+
+        If 'x' and 'y' coordinates already exist then they will not be added.
+
+        Args:
+            data_arr (xarray.DataArray): data object to add x/y coordinates to
+            area (pyresample.geometry.AreaDefinition): area providing the
+                coordinate data.
+            crs (pyproj.crs.CRS or None): CRS providing additional information
+                about the area's coordinate reference system if available.
+                Requires pyproj 2.0+.
+
+        Returns (xarray.DataArray): Updated DataArray object
+
+        """
+        if 'x' in data_arr.coords and 'y' in data_arr.coords:
+            return data_arr
+
+        if hasattr(area, 'get_proj_vectors'):
+            x, y = area.get_proj_vectors()
+        elif hasattr(area, 'get_proj_vectors_dask'):
+            # older pyresample with dask-only method
+            x, y = area.get_proj_vectors_dask(CHUNK_SIZE)
```

Comment:
Should we take the opportunity to remove this call ?

Reply:
Done